design/baremetal-operator/disable-introspection-intereface.md

```diff
@@ -0,0 +1,96 @@
+- The option to update status data with a status annotation when inspection is
+  performed by some means other than the regular BMH flow
+- An interface that's consistent with existing proposals, in particular the
+  [re-inspection interface](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/re-introspection-interface.md)
+
+### Non-Goals
+
+- There are no plans for any mechanism to trigger or consume data from any out
+  of band inspection process, other than providing data via the existing
+  status annotation.
+
+## Proposal
+
+To align with the re-inspection API proposal, the `inspect.metal3.io`
+annotation will be reused, but with an optional additional suffix, similar
+to that described for the [reboot API](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/reboot-interface.md)
+
+The optional `inspect.metal3.io/disable` annotation will be used to describe
+the situation where we wish to disable the default inspection behavior.
+
+When the BMO finds this annotation, it will move directly from `Registering`
+[state](https://github.com/metal3-io/baremetal-operator/blob/master/docs/BaremetalHost_ProvisioningState.png)
+to `Ready` and not perform any inspection.
+
+In this case (similar to the reboot interface), when the suffixed form of
+`inspect.metal3.io/disable` is found, it will never be removed by the BMO.
+
+If both inspect.metal3.io and inspect.metal3.io/disable forms are in specified,
+the `inspect.metal3.io/disable` form will take priority and the
+`inspect.metal3.io` annotation will be ignored.
+
+When `baremetalhost.metal3.io/status` is updated, any data should be
+consumed even in the `Ready` or `Provisioned` state, so that in the
+case where inspection data is collected via a live-image we can still
+store that data in the BMH status field.
+
+### User stories
+
+#### Fast-boot appliance live-image
+
+Where the LiveImage booted is an appliance that must be running as quickly as
+possible, it may be desirable to skip the time taken for inspection
+(and also cleaning which is discussed in an [existing proposal](https://github.com/metal3-io/metal3-docs/pull/151)
+
+#### Live-image installer does inspection
+
+[Installer ISO images](https://docs.fedoraproject.org/en-US/fedora-coreos/bare-metal/#_installing_from_live_iso)
+may be booted which can include their own inspection tooling.
+
+In this case, it is desirable to avoid the extra reboot and have the live-iso
+collect the required data (and update the BMH via the status annotation).
+
+## Open Questions
+
+When `baremetalhost.metal3.io/status` is updated and the BMH is in the `Ready`
+state, should we attempt to match profiles based on that inspection data?
+
+Given that the primary use-case for this is live-boot images (where no disk
+image is written to disk), and that profile matching is no longer the preferred
+interface for specifying root-disk hints, it may be reasonable to skip this?
```

Comment:
I think so, yes. We want to remove the profile part of the API, but don't want to make that a prerequisite to this work.